packages/valory/skills/decision_maker_abci/states/decision_receive.py

```diff
@@ -62,19 +63,27 @@ def end_block(self) -> Optional[Tuple[SynchronizedData, Enum]]:
         synced_data, event = cast(Tuple[SynchronizedData, Enum], res)
 
         if event == Event.DONE:
-            decision_receive_timestamp = self.most_voted_payload_values[-1]
+            decision_receive_timestamp = self.most_voted_payload_values[
+                -2
+            ]  # -1 is the should_be_sold, -2 is the decision_receive_timestamp
```

Comment:
I know this logic was already in place, but it's very risky. Let's please open an issue to permanently fix this, i.e., by instantiating a payload and accessing the corresponding property.